presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -421,6 +422,11 @@ public void addFunctionSignatures(IdentityLinkedHashMap<FunctionCall, Signature>
         return unmodifiableSet(columnReferences.keySet());
     }
 
+    public Map<Expression, FieldId> getColumnReferenceFields()
```

Comment:
At lower level... currently there are 5 places where we need `Set<Expression> columnReferences` and I didn't want to make them call `getColumnReferences().keySet()`. That's why i left `getColumnReferences()` method and added `getColumnReferenceFields()`. I'm not proud of the name though.